### PR TITLE
Add Examples app entry for a data-driven, anchored annotation example

### PR DESCRIPTION
**Pull request by avi-c**
### Summary of changes

This PR adds a new top-level example to the Examples app. This example shows how to add a custom SymbolLayer that supports adding data-driven image annotations that support anchoring on either the left, center, or right side of the annotation image. Annotations support stretchable images that will dynamically resized based on their text contents.

Custom Symbols

### CODE REVIEW

File: Examples/Examples/All Examples/CustomSymbolAnnotationsExample.swift [hunk 1]
```diff
@@ -0,0 +1,339 @@
+import UIKit
+import MapboxMaps
+import Turf
+
+@objc(CustomSymbolAnnotationsExample)
```
Comment (macdrevx) [inline Examples/Examples/All Examples/CustomSymbolAnnotationsExample.swift:5]: this should immediately precede the view controller class declaration

**Comment by macdrevx** [comment]
Heads-up that #187 will relocate the Examples project file & directory. If it merges first, some minor adjustments will be required here.

File: Examples/Examples/All Examples/CustomSymbolAnnotationsExample.swift (hunk 1 above)
Comment (avi-c) [reply Examples/Examples/All Examples/CustomSymbolAnnotationsExample.swift:5]: So you are suggesting I move the `enum` and `struct` declarations to the bottom of the file below the `CustomSymbolAnnotationsExample` class?

**Comment by avi-c** [comment]
Moved new example to the updated location in the codebase, along with it's image resources.

File: Examples/Examples/All Examples/CustomSymbolAnnotationsExample.swift (hunk 1 above)
Comment (avi-c) [reply Examples/Examples/All Examples/CustomSymbolAnnotationsExample.swift]: Actually, I believe I know what you meant and moved just the `@objc(CustomSymbolAnnotationsExample)` line to immediately precede the view controller class declaration.

**Comment by macdrevx** [comment]
@jmkiley how does this align with our plan for examples?

**Comment by avi-c** [comment]
@julianrex - Seems reasonable. How about "marker" instead of annotation?

File: Apps/Examples/Examples/All Examples/CustomSymbolAnnotationsExample.swift [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+            DebugFeature(coordinate: CLLocationCoordinate2DMake(40.716281, -74.004526), highlighted: true, sortOrder: 1, tailPosition: .right, label: "Broadway & Worth - Right Highlighted", imageName: "AnnotationRightHanded-Highlighted"),
+            DebugFeature(coordinate: CLLocationCoordinate2DMake(40.710194, -74.004248), highlighted: true, sortOrder: 1, tailPosition: .left, label: "Spruce & Gold - Left Highlighted", imageName: "AnnotationLeftHanded-Highlighted"),
+            DebugFeature(coordinate: CLLocationCoordinate2DMake(40.7128, -74.0060), highlighted: true, sortOrder: 2, tailPosition: .center, label: "City Hall - Centered Highlighted", imageName: "AnnotationCentered-Highlighted"),
+            DebugFeature(coordinate: CLLocationCoordinate2DMake(40.711427, -74.008614), highlighted: false, sortOrder: 3, tailPosition: .center, label: "Broadway & Vesey - Centered Stem", imageName: "AnnotationCentered")
+        ]
+
+        for (index, feature) in featureList.enumerated() {
+            var featurePoint = Feature(Point(feature.coordinate))
+
+            // set the feature attributes which will be used in styling the symbol style layer
```
Comment (jmkiley) [inline Apps/Examples/Examples/All Examples/CustomSymbolAnnotationsExample.swift:189]: nit: We should be consistent about comment structure. Could you update comments to use sentence casing with periods at the end?

**Review by jmkiley — CHANGES_REQUESTED** [review]
Thank you so much for the PR. Overall, most of my feedback is around code comments. Would you mind adding a few more comments in the example to highlight what is happening at each step?

File: Apps/Examples/Examples/Models/Examples.swift [hunk 3]
```diff
@@ -39,6 +39,9 @@ public struct Examples {
         Example(title: "Select an annotation",
                 description: "Select an annotation with a tap gesture.",
                 type: SelectAnnotationExample.self),
+        Example(title: "Add anchored annotations with a custom map layer.",
+                description: "Add anchored image annotations to the map with a custom SymbolLayer. Annotations are styled via feature attributes.",
```
Comment (jmkiley) [inline Apps/Examples/Examples/Models/Examples.swift:43]:
```suggestion
                description: "Add custom image annotations to the map with a SymbolLayer. The custom annotations are styled based on feature attributes and are anchored to specific points on the map.",
```

**Review by JCK-cloudLV — APPROVED** [review]

**Pull request closed without merging**